homeassistant/components/sonos/media_player.py

```diff
@@ -500,9 +500,14 @@ def update_media(self, event: SonosEvent | None = None) -> None:
         if new_status == "TRANSITIONING":
             return
 
-        self._play_mode = (
-            variables["current_play_mode"] if variables else self.soco.play_mode
-        )
+        if variables and "transport_state" in variables:
```

Comment:
Should we verify `current_track_uri` is in there as well or will there never be a case where `current_play_mode` is but `current_track_uri` isn't?

Reply:
Good question. It's been all-or-nothing in all the examples I've collected. Any transport error will populate `variables` with only `transport_status`, `transport_error_description`, and `transport_error_uri`.